### PR TITLE
Modular GW.js Support 

### PR DESCRIPTION
**Pull request by damassi**
This fits https://github.com/thegroundwork/groundwork.js/pull/12 into the cleanup work located at https://github.com/thegroundwork/groundwork.js/pull/13. 

Additionally, a few changes:
- Renamed `dist` to `lib` for idiomatic and clarity reasons: `import Events from 'groundwork.js/lib/modules/Events'`.
- Moved tests in with sources under a `src/__test__` folder.
- Added new `--modules` build path in `webpack.config`
- Added a new `modules` example, which still needs to be built out with a UI. Right now it just outputs the result of a modular instantiation into the console. 

**To run:**
- `npm install && npm run example:modules`

@DarrenN - There might be a few loose details here and there that I need to address but this is largely ready to review.

**TODO:**
- [x] Style `modules` example
- [x] Update `Makefile` to move `lib` to `latest` over s3 rather than in project so that we have better modular imports 
- [x] Update / cleanup README.md
- [x] Flip keys and CDN back to PROD (cdn, cloudfront travis)

Closes https://github.com/thegroundwork/groundwork.js/pull/12.

### CODE REVIEW

File: examples/modules/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  "babel": {
+    "presets": [
+      "es2015",
+      "stage-0"
+    ]
+  },
+  "devDependencies": {
+    "babel-preset-es2015": "6.3.13",
+    "babel-preset-stage-0": "6.5.0",
+    "react-scripts": "0.2.1"
```
Comment (DarrenN) [inline examples/modules/package.json]: wut dis

File: examples/modules/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  "dependencies": {
+    "react": "^15.2.1",
+    "react-dom": "^15.2.1"
```
Comment (DarrenN) [inline examples/modules/package.json]: why is react a dependency of the project?

**Comment by DarrenN** [comment]
I see there is now a React example. Is there still a Non-React example?

**Comment by DarrenN** [comment]
For some reason `npm watch-dev` doesn't start the server. It hangs somewhere during the webpack compile:

screenshot 2016-08-11 09 29 54

You'll notice in the activity monitor that Node isn't doing anything, it looks like the process just stopped somewhere in the pipeline. I blew away `node_modules` and started over, same thing, it stops right here:

```
/Users/darren/.nvm/versions/node/v4.4.7/lib/node_modules/groundwork.js -> /Users/darren/github/groundwork.js
NODE_ENV=development node_modules/.bin/webpack -w
Hash: 3bce21322dc300d12243
Version: webpack 1.13.0
Time: 5139ms
                  Asset     Size  Chunks             Chunk Names
          groundwork.js   619 kB       0  [emitted]  index
      groundwork.js.map   744 kB       0  [emitted]  index
             index.html   3.9 kB          [emitted]
    examples/index.html  3.08 kB          [emitted]
     examples/auth.html  8.18 kB          [emitted]
examples/supporter.html  9.17 kB          [emitted]
   [0] multi index 28 bytes {0} [built]
    + 340 hidden modules
Child html-webpack-plugin for "index.html":
        + 4 hidden modules
Child html-webpack-plugin for "examples/index.html":
        + 4 hidden modules
Child html-webpack-plugin for "examples/auth.html":
        + 4 hidden modules
Child html-webpack-plugin for "examples/supporter.html":
        + 4 hidden modules
```

My env:

```
$ npm -v // 3.10.5
$ node -v // v4.4.7
```

**Comment by DarrenN** [comment]
When I run the examples (ex: `cd examples/modules && npm install && npm start`) I didn't get a prompt about ports, my browser was automatically opened to `3000`:

screenshot 2016-08-11 09 39 55

And my terminal went to:

screenshot 2016-08-11 09 40 17

There was nothing at `http://localhost:3001/`

That said, when I popped open console on `3000` I did the see the modular builds messages:

screenshot 2016-08-11 09 41 51

🎉

**Comment by DarrenN** [comment]
Q: Why are we running `eslint-flowtype` in the examples? I ask because it immediately tossed an error in Emacs because I don't have it globally installed.

**Comment by DarrenN** [comment]
Q: I'm looking at the compiled modules in `/lib` while running the examples and the individual modules look _crazy_ large:

screenshot 2016-08-11 10 11 06

Is this just an artifact of the dev builds or is something amiss?

**Comment by DarrenN** [comment]
If we're going to drop the tests into the `/src` folder I think I'd prefer they lived in

**Comment by DarrenN** [comment]
Was able to get the non-React examples up and running with `npm start`. We'll need to adjust those examples to use the newer builds:

```
Uncaught TypeError: Groundwork is not a constructor
```

from Line 103 in `index.html`:

```
 var gw = new Groundwork();
```

**Comment by damassi** [comment]
@DarrenN - Answers inline:

> I see there is now a React example. Is there still a Non-React example?

Yup, the "simple" examples are still the same, meant to illustrate using GW.js without needing a full app setup.

> For some reason npm watch-dev doesn't start the server. It hangs somewhere during the webpack compile:

This is because it doesn't actually start a server, but a watcher on webpack (not webpack dev server). Why? Because when we're developing examples (which have their own server) all we need is the ability to recompile on the source-code side so that the example picks it up. I will clarify this process today.

> When I run the examples (ex: cd examples/modules && npm install && npm start) I didn't get a prompt about ports, my browser was automatically opened to 3000

_bleary eyed_ 

> Why are we running eslint-flowtype in the examples? I ask because it immediately tossed an error in Emacs because I don't have it globally installed.

This is part of the cool `create-react-app` util that we're using on the examples side. It shouldn't be asking for anything global; let me research

> Q: I'm looking at the compiled modules in /lib while running the examples and the individual modules look crazy large

Will be optimizing this today.

> If we're going to drop the tests into the /src folder I think I'd prefer they lived in a __test__ folder as we do in other projects.

Will do. I agree it looks a little cluttered, but was trying to avoid the '../' in the test imports and was curious what you thought. NBD though, will update.

**Comment by damassi** [comment]
@DarrenN (moved this from slack to here) - can you do me a favor and pull down latest on gw.js and then run `make run-examples` and see if everything boots ok (should see window pop open at `localhost:3000` and  then go into example source and see if any linter errors shown in emacs?

**Comment by damassi** [comment]
> Was able to get the non-React examples up and running with npm start. We'll need to adjust those examples to use the newer builds:

This was due to me exporting as ES6 modules

```javascript
// Before
export default Groundwork;

export {
  groundworkFactory
};

// Then, in html page
console.log(Groundwork) // => {_esModule: true, default}
```

Switching back to 
```javascript
module.exports.groundworkFactory = groundworkFactory;
module.exports = Groundwork;
```
Fixed it.

**Comment by DarrenN** [comment]
@damassi `make run-examples` ran fine and asked me for a new port as I already had 3000 in use. Works.

**Comment by damassi** [comment]
Awesome, final PR coming up in a sec

**Comment by DarrenN** [comment]
Re: flow-type

You misunderstand, because the flowtype plugin is defined somewhere (I'm guessing in an `.eslintrc` or similar) my local eslint wants to use it, but can't because I need it to be globally installed -> https://github.com/syl20bnr/spacemacs/blob/a298a5a311dfe8b449b5f538740303f1ab3d5695/layers/%2Bframeworks/react/README.org

**Comment by damassi** [comment]
I thought i disabled eslint via a change i made in package.json -- did you try to see if it still threw a problem?

**Comment by damassi** [comment]
Alright @DarrenN - this is good to go, and got the module sizes down and README.md updated:

screen shot 2016-08-11 at 4 56 53 pm

Can you give it one last pass and let me know? (Note that i'll be out till Monday.)

**Comment by DarrenN** [comment]
Looks great @damassi - :+1: - lets talk on Monday before merging just to cover everything.

**Pull request merged**